prefect_census/client.py

```diff
@@ -0,0 +1,93 @@
+"""Module containing client for interacting with the Census API"""
+from httpx import AsyncClient, Response
+
+
+class CensusClient:
+    """
+    Client for interacting with the Census API.
+
+    Args:
+        api_key: API key to authenticate with the Census API.
+    """
+
+    def __init__(self, api_key: str):
+        self._closed = False
+        self._started = False
+
+        self.client = AsyncClient(
+            base_url="https://app.getcensus.com",
+            headers={"Authorization": f"Bearer {api_key}"},
+        )
+
+    async def call_endpoint(
+        self,
+        http_method: str,
+        path: str,
+    ) -> Response:
```

Comment:
To support API endpoints that we don't have specific methods for, I think we'll want to include parameters for query params and a request body as well.
```suggestion
    async def call_endpoint(
        self,
        http_method: str,
        path: str,
        params: Optional[Dict[str, Any] = None,
        json: Optional[Dict[str, Any] = None,
    ) -> Response:
```